test/e2e/samples_test.go

```diff
@@ -0,0 +1,127 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package e2e
+
+import (
+	"bufio"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	apmtype "github.com/elastic/cloud-on-k8s/pkg/apis/apm/v1beta1"
+	commonv1beta1 "github.com/elastic/cloud-on-k8s/pkg/apis/common/v1beta1"
+	estype "github.com/elastic/cloud-on-k8s/pkg/apis/elasticsearch/v1beta1"
+	kbtype "github.com/elastic/cloud-on-k8s/pkg/apis/kibana/v1beta1"
+	"github.com/elastic/cloud-on-k8s/test/e2e/test"
+	"github.com/elastic/cloud-on-k8s/test/e2e/test/apmserver"
+	"github.com/elastic/cloud-on-k8s/test/e2e/test/elasticsearch"
+	"github.com/elastic/cloud-on-k8s/test/e2e/test/kibana"
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/apimachinery/pkg/util/rand"
+	"k8s.io/apimachinery/pkg/util/yaml"
+)
+
+func TestSamples(t *testing.T) {
+	sampleFiles, err := filepath.Glob("../../config/samples/*/*.yaml")
+	require.NoError(t, err, "Failed to find samples")
+
+	decoder := createDecoder(t)
+	for _, sample := range sampleFiles {
+		builders := createBuilders(t, decoder, sample)
+		testName := mkTestName(t, sample)
+		t.Run(testName, func(t *testing.T) {
+			test.Sequence(nil, test.EmptySteps, builders...).RunSequential(t)
+		})
+	}
+}
+
+func createDecoder(t *testing.T) runtime.Decoder {
+	t.Helper()
+
+	scheme := runtime.NewScheme()
+	scheme.AddKnownTypes(estype.GroupVersion, &estype.Elasticsearch{}, &estype.ElasticsearchList{})
+	scheme.AddKnownTypes(kbtype.GroupVersion, &kbtype.Kibana{}, &kbtype.KibanaList{})
+	scheme.AddKnownTypes(apmtype.GroupVersion, &apmtype.ApmServer{}, &apmtype.ApmServerList{})
+	return serializer.NewCodecFactory(scheme).UniversalDeserializer()
+}
+
+func mkTestName(t *testing.T, path string) string {
+	t.Helper()
+
+	baseName := filepath.Base(path)
+	baseName = strings.TrimSuffix(baseName, ".yaml")
```

Comment:
Is there a limitation on having `.yaml` in the test name or is this more of a stylistic choice?

Reply:
Purely stylistic choice.